frontend/README.md

```diff
@@ -0,0 +1,151 @@
+
+If you can't turn on websockets, you will see errors in the console (that's not very impacting) and you'll have to reload your django server regularly, because polling requests don't close, and after so many web page reloads, the django process won't be able to accept new requests.
+
+
+## Type checking
+
+Our code uses Flow to enforce type checking. This is a good way to significantly improve resilience to bugs, and it removes some burden from unit tests (because Flow ensures that we use functions and components correctly).
+
+To check for Flow issues during development while you edit files, run:
+
+    yarn flow:dev
+
+To learn more, you can read [Why use static types in JavaScript?](https://medium.freecodecamp.org/why-use-static-types-in-javascript-part-1-8382da1e0adb) or the official [Flow documentation](https://flow.org/en/docs/). Additionally, you can read through the [web-ext guide](https://github.com/mozilla/web-ext/blob/master/CONTRIBUTING.md#check-for-flow-errors) for hints on how to solve common Flow errors.
+
+Until we define our set of rules, please refer to the [addons team's Flow manifesto](https://github.com/mozilla/addons-frontend/#flow) regarding specific usage and edge cases.
+
+
+## Testing
+
+Tests are run using [`jest`](https://facebook.github.io/jest/). We use [`enzyme`](http://airbnb.io/enzyme/docs/api/) for mounting React components and [`sinon`](http://sinonjs.org/) for mocking.
+
+To run the test suite, use:
+
+    $ yarn test
+
+It will start an auto-reloading test runner, that will refresh every time you make a change to the code or tests.
+
+Tests are put in files called `fileToTest.test.js` in the same directory as the file to test. Inside test files, test suites are created. There should be one test suite per component, using this notation:
+
+```javascript
+describe('<Component>', () => {
+    // test suite here
+});
+```
+
+Individual tests follow `mocha`'s syntax:
+
+```javascript
+it('does something', () => {
+    // unit test here
+});
+```
+
+We use `jest`'s [`expect`](https://facebook.github.io/jest/docs/en/expect.html) assertion tool.
+
+
+## Development resources
+
+### Integration between Django and React
+
+- [Making React and Django play well together — Fractal Ideas](https://fractalideas.com/blog/making-react-and-django-play-well-together/)
+- [Making React and Django play well together - the “hybrid app” model — Fractal Ideas](https://fractalideas.com/blog/making-react-and-django-play-well-together-hybrid-app-model/)
+
+### Code architecture
+
+- [Three Rules For Structuring (Redux) Applications — Jack Hsu](https://jaysoo.ca/2016/02/28/organizing-redux-application/)
+- [The Anatomy Of A React & Redux Module (Applying The Three Rules) — Jack Hsu](https://jaysoo.ca/2016/02/28/applying-code-organization-rules-to-concrete-redux-code/)
+- [Additional Guidelines For (Redux) Project Structure — Jack Hsu](https://jaysoo.ca/2016/12/12/additional-guidelines-for-project-structure/)
+
+### Tools documentation
```

Comment:
That's already covered on top of the page. Remove?

Reply:
These point directly to the documentation of the tools, I think it's not a problem to have them kind of doubled because I expect it will be convenient at least to me in the future. :)